src/Xamarin.Android.Build.Tasks/Tasks/ResolveSdksTask.cs

```diff
@@ -423,13 +423,21 @@ bool ValidateApiLevels ()
 
 			if (UseLatestAndroidPlatformSdk) {
 				AndroidApiLevel         = GetMaxInstalledApiLevel ().ToString ();
-				SupportedApiLevel       = GetMaxSupportedApiLevel (AndroidApiLevel);
+				SupportedApiLevel       = GetMaxStableApiLevel ().ToString ();
 				int maxInstalled, maxSupported = 0;
 				if (int.TryParse (AndroidApiLevel, out maxInstalled) && int.TryParse (SupportedApiLevel, out maxSupported) && maxInstalled > maxSupported) {
 					Log.LogDebugMessage ($"API Level {AndroidApiLevel} is greater than the maximum supported API level of {SupportedApiLevel}. " +
 						"Support for this API will be added in a future release.");
 					AndroidApiLevel = SupportedApiLevel;
 				}
+				if (!string.IsNullOrWhiteSpace (TargetFrameworkVersion)) {
+					var userSelected = MonoAndroidHelper.SupportedVersions.GetApiLevelFromFrameworkVersion (TargetFrameworkVersion);
+					// overwrite using user version only if it is 
+					// above the maxStableApi and a valid apiLevel.
+					if (userSelected != null && userSelected > maxSupported && userSelected <= maxInstalled) {
```

Comment:
None of this make sense. :-(

The use case is this: "soon", Google will presumably release API-P in alpha form, for release as API-28 this fall.

When we provide a binding, it will *probably* be `$(TargetFrameworkVersion)`=v8.99.

Assuming the user has `v8.99` installed -- *how* that's done is "elsewhere" -- the user should be able to opt in to using it by setting `$(TargetFrameworkVersion)`=v8.99.

Thus, `!string.IsNullOrWhiteSpace (TargetFrameworkVersion)` (line 433) will be `true`.

`userSelected` (line 434), meanwhile, will be `null`, as *there is no API level for unstable APIs*.  Relatedly, `MonoAndroidHelper.SupportedVersions.GetIdFromFrameworkVersion("v8.99")` will return `"P"`.

Similarly, `maxInstalled` comes from `int.TryParse()` comes from `GetMaxInstalledApiLevel()`, which only looks at values which can be parsed into an `int`. As such, *API-P will not be present or used*, meaning -- for our current exercise -- `AndroidApiLevel` is ~useless. Consequently, `maxInstalled` is likewise useless.

In short, I don't see how this will do what we want it to do. :-(